keps/0002-dynamic-crds.md

```diff
@@ -0,0 +1,68 @@
+---
+kep-number: draft-20190214
+title: Dynamic CRDs
+authors:
+  - "@gerred"
+owners:
+  - TBD
+editor: TBD
+creation-date: 2019-02-14
+last-updated: 2019-02-14
+status: provisional
+---
+
+# Framework-specific Dynamic Custom Resource Definitions
+
+## Table of Contents
+
+- [Framework-specific Dynamic Custom Resource Definitions](#framework-specific-dynamic-custom-resource-definitions)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Implementation History](#implementation-history)
+
+## Summary
+
+This KEP aims to make the end user experience for frameworks more specific to the business domain they represent. By implementing dynamic CRDs, frameworks will be able to represent their components in a declarative way, specific to their framework. Additionally, it enables framework developers to provide day 2 operations tasks as CRDs, complete with their own plans and tasks.
+
+## Motivation
+
+Currently, the interface for using frameworks in KUDO is very generic. Users create an `Instance` CRD with specs. Operator developers and users expect to be able to use contextual business objects for their operators instead of generic objects. This enables a more focused experience for users of KEP.
+
+The goal of this KEP is to improve the end user UX through dynamic CRDs. Other than the ability to specfiy CRDs, and adjusting existing framework development CRDs to accomodate this change, it is not the goal of this KEP to change the framework development UX.
+
+### Goals
+
+- Create a mechanism for framework developers to specify a CRD
+- Enable management for custom resources based on dynamic CRDs. Deploying a framework specific custom resource should deploy a plan as `Instance` was able to before.
```

Comment:
Maybe something more like:

Instantiating a FrameworkVersion as a unique CRD instead of a generic Instance CRD